packages/synapse-react-client/src/components/CardContainer/CardContainer.integration.test.tsx

```diff
@@ -0,0 +1,147 @@
+import { render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import React from 'react'
+import { SynapseConstants } from '../../utils'
+import CardContainer, { CardContainerProps } from './CardContainer'
+import { QueryVisualizationWrapper } from '../QueryVisualizationWrapper/QueryVisualizationWrapper'
+import { createWrapper } from '../../testutils/TestingLibraryUtils'
+import { QueryBundleRequest } from '@sage-bionetworks/synapse-types'
+import syn16787123Json from '../../mocks/query/syn16787123'
+import {
+  DEFAULT_PAGE_SIZE,
+  MEDIUM_USER_CARD,
+} from '../../utils/SynapseConstants'
+import mockUserCardTableQueryResultBundle from '../../mocks/query/mockUserCardTableQueryResultBundle'
+import { server } from '../../mocks/msw/server'
+import { mockUserProfileData } from '../../mocks/user/mock_user_profile'
+import { QueryWrapper } from '../QueryWrapper'
+import { getHandlersForTableQuery } from '../../mocks/msw/handlers/tableQueryHandlers'
+import { cloneDeep } from 'lodash-es'
+import { QueryContextConsumer } from '../QueryContext'
+import { InfiniteQueryContextType } from '../QueryContext/QueryContext'
+
+const sql = 'SELECT * FROM syn16787123'
+
+const lastQueryRequest: QueryBundleRequest = {
+  concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+  entityId: 'syn16787123',
+  partMask: 255,
+  query: {
+    sql,
+    limit: DEFAULT_PAGE_SIZE,
+    offset: 0,
+  },
+}
+
+const dataWithOnePage = cloneDeep(syn16787123Json)
+dataWithOnePage.queryCount =
+  dataWithOnePage.queryResult!.queryResults.rows.length
+dataWithOnePage.queryResult!.nextPageToken = undefined
+
+const dataWithMultiplePages = cloneDeep(syn16787123Json)
+dataWithMultiplePages.queryCount =
+  dataWithMultiplePages.queryResult!.queryResults.rows.length + 1
+dataWithMultiplePages.queryResult!.nextPageToken = {
+  token: 'abcd',
+}
+
+const unitDescription = 'study'
+
+let capturedQueryContext: InfiniteQueryContextType | undefined
+
+const renderComponent = (props: CardContainerProps) => {
+  return render(
+    <QueryWrapper initQueryRequest={lastQueryRequest} isInfinite={true}>
+      <QueryContextConsumer>
+        {queryContext => {
+          capturedQueryContext = queryContext as InfiniteQueryContextType
+          return (
+            <QueryVisualizationWrapper unitDescription={unitDescription}>
+              <CardContainer {...props} />
+            </QueryVisualizationWrapper>
+          )
+        }}
+      </QueryContextConsumer>
+    </QueryWrapper>,
```

Comment:
Most of these test file changes have the same tests, but now
1. Use QueryWrapper and mock table query endpoints using mock-service-worker instead of creating our own QueryContext object, which exposed too many implementation details that broke on a refactor
2. Move test file to `src/` folder to be fully type-checked